flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -780,6 +780,22 @@ ANY, and(logical(LogicalTypeRoot.BOOLEAN), LITERAL)
                     .outputTypeStrategy(TypeStrategies.aggArg0(t -> t, true))
                     .build();
 
+    public static final BuiltInFunctionDefinition LEAD =
+            BuiltInFunctionDefinition.newBuilder()
+                    .name("lead")
+                    .kind(AGGREGATE)
+                    .inputTypeStrategy(SpecificInputTypeStrategies.LEAD_LAG)
+                    .outputTypeStrategy(SpecificTypeStrategies.LEAD_LAG)
```

Comment:
for our book keeping add:
```
.runtimeProvided()
```

Reply:
Apparently, it has runtime implications not just bookkeeping.

The way I understand it from the failing test, if `runtimeProvided` is used the `BuiltinFunctionDefinition` will be converted to `BridgingSqlAggFunction` instead of calcite's `SqlLeadLagFunction` which we depend on in lots of places including code generation.

I reverted it back. Please let me know wdyt.

Reply:
Or was this `runtimeDeferred`? I don't remember what was the difference.

Reply:
Yep, I can see `runtimeDeferred` does nothing:

```
        /**
         * Specifies that this {@link BuiltInFunctionDefinition} will be mapped to a Calcite
         * function.
         */
        public Builder runtimeDeferred() {
            // This method is just a marker method for clarity. It is equivalent to calling
            // neither {@link #runtimeProvided} nor {@link #runtimeClass}.
            return this;
        }

```